File: server/src/main/java/com/fluffy/exam/ui/request/PublishExamWebRequest.java
package com.fluffy.exam.ui.request;

import com.fluffy.exam.application.request.PublishExamRequest;
import com.fluffy.exam.application.request.question.QuestionRequest;
import com.fluffy.global.web.Accessor;
import java.util.List;

public record PublishExamWebRequest(List<QuestionRequest> questions) {
    public PublishExamRequest toAppRequest(Long examId, Accessor accessor) {
        return new PublishExamRequest(examId, questions, accessor);
    }
}

File: server/src/main/kotlin/com/fluffy/exam/ui/request/PublishExamWebRequest.kt
package com.fluffy.exam.ui.request

import com.fluffy.exam.application.request.PublishExamRequest
import com.fluffy.exam.application.request.question.QuestionRequest
import com.fluffy.global.web.Accessor

data class PublishExamWebRequest(val questions: List<QuestionRequest>) {

    fun toAppRequest(examId: Long, accessor: Accessor): PublishExamRequest {
        return PublishExamRequest(
            examId = examId,
            questions = questions,
            accessor = accessor
        )
    }
}
